simphony/tools/maxwell_equal_area_rule.py

```diff
@@ -0,0 +1,240 @@
+
+
+def _foff(arg1, arg2, f, C):
+    return f(arg1, arg2) - C
+
+
+def _finters(a, b, f, arg2, C):
+    """Find an intersection of a 2 argument function f and a constant C.
+
+    Parameters
+    ----------
+    a : lower limit for the search interval
+    b : upper limit for the search interval
+    f : function of 2 arguments
+    arg2 : second argument of f
+    C : constant
+
+    Returns
+    -------
+    The intersection
+
+    Raises
+    ------
+    RuntimeError : If an intersection is not found.
+
+    """
+    # brentq from scipy
+    x0, r = brentq(_foff, a, b, (arg2, f, C), full_output=True)
+
+    if not r.converged:
+        raise RuntimeError("An intersection not found: abort!")
+
+    return x0
+
+
+def _area_calc(Pr, Tr, reos, head_ab, body_ab, tail_ab):
+    """Compute the Maxwell's areas.
+
+    That is, compute the areas demarcated
+    by the isotherm and a reduced pressure.
+
+    Parameters
+    ----------
+    Pr : reduced pressure, aka constant pressure
+    Tr : reduced temperature
+    reos : reduced equation of state
+    head_ab : head part of the isotherm, aka head
+    body_ab : body part of the isotherm, aka body
+    tail_ab : tail part of the isotherm, aka tail
+
+    Returns
+    -------
+    A tuple with two elements:
+        1. A tuple including the intersections between the constant
+           pressure and the head (1.1.), body (1.2.), and tail (1.3.) part.
+
+        2. A tuple including the area
+           2.1. below the constant pressure (negative),
+           2.2. above the constant pressure (positive).
+
+    """
+    h0 = _finters(head_ab[0], head_ab[1], reos, Tr, Pr)
+    b0 = _finters(body_ab[0], body_ab[1], reos, Tr, Pr)
+    t0 = _finters(tail_ab[0], tail_ab[1], reos, Tr, Pr)
+
+    # quad from scipy
+    a1 = quad(reos, h0, b0, (Tr,))[0] - (b0 - h0)*Pr
+    a2 = quad(reos, b0, t0, (Tr,))[0] - (t0 - b0)*Pr
+
+    return ((h0, b0, t0), (a1, a2))
+
+
+def _area_diff(Pr, Tr, reos, head_ab, body_ab, tail_ab):
+    """Compute the difference between Maxwell's areas.
+
+    That is, compute the difference between areas
+    demarcated by the isotherm and a reduced pressure.
+
+    Parameters
+    ----------
+    Pr : reduced pressure, aka constant pressure
+    Tr : reduced temperature
+    reos : reduced equation of state
+    head_ab : head part of the isotherm, aka head
+    body_ab : body part of the isotherm, aka body
+    tail_ab : tail part of the isotherm, aka tail
+
+    Returns
+    -------
+    The difference between Maxwell's areas (absolute value).
+
+    """
+    (intersects, areas) = _area_calc(Pr, Tr, reos, head_ab, body_ab, tail_ab)
+    return np.fabs(areas[0] + areas[1])
+
+
+def liquid_vapour_density(Trs, reos, iSearchMax=40.0, RhorMin=1e-12):
+    """Compute liquid-vapour densities for the given reduced temperatures.
+
+    The densities are computed by utilizing the Maxwell's equal area rule.
+
+    Parameters
+    ----------
+    Trs : Reduced temperatures (iterable)
```

Comment:
I think it's better to define parameter types explicitly (iterable of int, float, string etc.) and add a description of its use on the next line. In this case it could be
"""
   Trs : iterable of float
           Reduced temperatures
"""